File: app/android/app/src/main/java/bosch/smartcampus/thermalcomfortstudy/packet/SkinTemperatureItem.java
package bosch.smartcampus.thermalcomfortstudy.packet;

import bosch.smartcampus.thermalcomfortstudy.lib.Timestamp;

/**
 * Created by rsukkerd on 6/27/16.
 */
public class SkinTemperatureItem extends BandDataItem {
    private static final String SKIN_TEMPERATURE_NAME = "SkinTemperature";

    private double skinTemperature;

    public SkinTemperatureItem(String username, Timestamp timestamp,
                               double skinTemperature) {
        super(username, timestamp);
        this.skinTemperature = skinTemperature;
    }

    public double getSkinTemperature() {
        return skinTemperature;
    }

    @Override
    public String toXML() {
        StringBuilder builder = new StringBuilder();
        builder.append("<" + ELEMENT + " id=\"" + getId() + "\">");
        builder.append(getTransducerDataElement(SKIN_TEMPERATURE_NAME, Double.toString(skinTemperature)));
        builder.append("</" + ELEMENT + ">");
        return builder.toString();
    }
}

File: app/iOS/Thermal Comfort Study/SkinTemperatureItem.swift
//
//  SkinTemperatureItem.swift
//  Thermal Comfort Study
//
//  Created by Madhav Achar on 10/14/16.
//  Copyright © 2016 boschsc. All rights reserved.
//

import Foundation

class SkinTemperatureItem: BandDataItem {
    
    private static let SKIN_TEMPERATURE_NAME = "SkinTemperature"
    private var skinTemperature: Double!
    
    init(username: String, timestamp: Date, skinTemperature: Double) {
        super.init(username, timestamp)
        self.skinTemperature = skinTemperature
    }
    
    func toXML() -> String {
        return super.toXML(dataName: type(of: self).SKIN_TEMPERATURE_NAME,
                           value: self.skinTemperature)
    }
}
